### ISSUE TITLE
Infinite reloading behaviour

### ISSUE THREAD

**Issue by happysalada** [phoenixframework/phoenix_live_view#1195]
### Environment

* Elixir version (elixir -v): 1.5.6
* Phoenix version (mix deps): 0.14.7
* Phoenix LiveView version (mix deps): 0.14.7
* NodeJS version (node -v): 15.0.1
* NPM version (npm -v): yarn 1.22.5
* Operating system: happening on a docker container running on Alpine 3.12.1
* Browsers you attempted to reproduce this bug on (the more the merrier):
firefox 81.0.2, Chrome 86.0.4240.114 
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no:
This problem happens on production on a clean build.

### Actual behavior

There is a page that keeps on reloading every 5~20 seconds. This page is a form, so the user would be typing something and then the page reloads.
This does not happen consistently, only sometimes, and never in development.

here is an example of the page it happens on.
https://union.scandinavianmarkets.com/regulated_entities/226/currencies/2/accounts/new?referrer_factsheet_link_id=4375f764-0bae-423c-9152-236d11196f34

I have checked the console messages, nothing appears. There is simply a full reload of the page. Looking at the network tab, it's simply a new connection to the page.

On the elixir side of things, nothing shows up in the logs, I just have the following pattern

```
2020-10-25 01:21:03 | {"image":"happysalada/union:2020-10-23_18.38.22","level":"info","message":"GET /regulated_entities/226/currencies/2/accounts/new"}
2020-10-25 01:20:46 | {"image":"happysalada/union:2020-10-23_18.38.22","level":"info","message":"GET /regulated_entities/226/currencies/2/accounts/new"}
2020-10-25 01:20:30 | {"image":"happysalada/union:2020-10-23_18.38.22","level":"info","message":"GET /regulated_entities/226/currencies/2/accounts/new"}
2020-10-25 01:20:13 | {"image":"happysalada/union:2020-10-23_18.38.22","level":"info","message":"GET /regulated_entities/226/currencies/2/accounts/new"}
2020-10-25 01:19:57 | {"image":"happysalada/union:2020-10-23_18.38.22","level":"info","message":"GET /regulated_entities/226/currencies/2/accounts/new"}
...
```

The application is deployed on a single node and going through reverse proxy (caddy).
There is nothing particular in the logs of the reverse proxy.
The node is not under particular load (17% CPU of 8 core intel CPU).
Memory usage is high but not that high (19 GB out of 64 GB).

here is my Endpoint configuration in case you need it
```
config :union, UnionWeb.Endpoint,
  cache_static_manifest: "priv/static/cache_manifest.json",
  check_origin: false,
  server: true,
  live_view: [signing_salt: "my_signing_salt_has_been_replaced"]
```

Here is some of the code I'm running
```
  @impl true
  def mount(params, session, socket) do
    case managing_entity(params) do
      {:ok, managing_entity} ->
        user = user_from(session)
        referrer_underlying = referrer_underlying(params)

        socket =
          socket
          |> assign(:changeset, Form.new_changeset())
          |> assign(:managing_entity, managing_entity)
          |> assign(:admin, admin?(user, managing_entity))
          |> assign(:countries, Countries.dropdown())
          |> assign(:currencies, currencies(params))
          |> assign(:current_user, user)
          |> assign(:referrer_details, referrer_details(referrer_underlying))
          |> assign(:theme, theme_for(referrer_underlying, managing_entity))

        {:ok, socket}

      {:not_found, _} ->
        {:ok,
         socket
         |> redirect(to: Routes.page_path(Endpoint, :not_found))}
    end
  end
```
with 2 handle events, one on validate and one on submit (it's just a form)
```
  @impl true
  def handle_event(
        "validate",
        %{"params" => params},
        socket
      ) do
    {:noreply,
     socket
     |> assign(:changeset, Form.validate(params) |> Map.put(:action, :validate))}
  end
```

I'm not sure how to debug further. I'm open to adding more logging or trying custom code.

Over the course of the single day it might not happen, but over the course of a single week, the problem will definitely show up.

Let me know if I have missed anything.

### Expected behavior

The page doesn't reload if it doesn't need to

**Comment by josevalim** [comment]
Hi @happysalada! Please try using LiveView master and see if the issue is still present. If not, then please open up a new issue with a minimal app that reproduces it. Your instructions here are clear but unfortunately there isn't much we can do trying to access a production app, we need to be able to reproduce it locally. Thanks!

**Comment by happysalada** [comment]
We have found the problem.

We were trying to load something too big on the page (terms and conditions from some provider that are ridiculously long). 

It seems there is a timeout with liveview that says that if it doesn't receive a response within a certain time, then reload the page. This problem is prominent only for users with a poor connection. In South Africa for example it was happening continuously, whereas in Japan it more rarely happening.

We have solved the problem by rendering the terms and conditions in a separate page that doesn't have liveview.

Just documenting here in case the same problem happens to someone else.
